File: src/internal/domain/utils/jwt/generate_token.rs
use chrono::Utc;
use jsonwebtoken::{encode, Algorithm, EncodingKey, Header};

use crate::dto::res::token::{token_claim::TokenClaim, token_user_response::TokenUserResponse};

pub async fn generate_token(jwt_secret_key: String, user: TokenUserResponse) -> Result<String, String> {
    // token algorithem
    let chosen_alogorithm = Algorithm::HS512;

    // encoding key from config
    let encoding_key = EncodingKey::from_secret(jwt_secret_key.as_bytes());

    // create timestamp
    let issued_at = Utc::now();

    // create expires at timestamp
    let expires_at = issued_at + chrono::Duration::hours(5);

    // token payload
    let claims = TokenClaim {
        sub: user.id.to_string(),
        email: user.email,
        name: user.name,
        exp: expires_at.timestamp() as usize,
        iat: issued_at.timestamp() as usize,
    };

    let header = Header::new(chosen_alogorithm);

    let token = match encode(&header, &claims, &encoding_key) {
        Ok(token) => token,
        Err(e) => return Err(format!("Failed to encode token: {}", e)),
    };

    Ok(token)
}

File: minitest/utils/jwt/generate_token.py
import jwt
from datetime import datetime, timedelta
from django.conf import settings

def create_jwt_token(user):
    """
    사용자 정보를 기반 JWT 토큰 생성
    """
    now = datetime.utcnow()
    expiration_time = now + timedelta(hours=5)
    
    payload = {
        "sub": str(user.id), 
        "email": user.email,
        "name": user.name,  
        "exp": expiration_time,
        "iat": now
    }
    
    token = jwt.encode(
        payload, 
        settings.JWT_SECRET_KEY, 
        algorithm='HS512' # SHA-512 해시 알고리즘
    )
    
    return token